Pp 1501 - Automate font-awesome resource generation

matthew-craig-ncino opened the pull request:
https://ncinodev.atlassian.net/browse/PP-1501

- I updated the circle.yml file to use the latest practices
- I removed as much as possible from the instructions
- I added the collection of the resource file INTO the artifacts should we ever want to collect them for future use.
- Build of master branch automatically generates a "release" binary named "font_awesom.resource" that can be used by CI or develoeprs to replace "src/staticresources/font_awesome.resource" in force framework

matthew-craig-ncino committed:
PP-1501 Updated circle.yml to latest practices

matthew-craig-ncino committed:
PP-1501 Added code to build components for font-awesome

matthew-craig-ncino committed:
PP-1501 escaped # signs

matthew-craig-ncino committed:
PP-1501 removed extra generated files

matthew-craig-ncino committed:
PP-1501 Testing build.sh

matthew-craig-ncino committed:
PP-1501 Added local instance of shell script call

matthew-craig-ncino committed:
PP-1501 Changed path

matthew-craig-ncino committed:
PP-1501 Added requird dirs

matthew-craig-ncino committed:
PP-1501 Bash execute build.sh

matthew-craig-ncino committed:
PP-1501 Cleanup

matthew-craig-ncino committed:
PP-1501 Removed build.sh copied resource to artifacts

matthew-craig-ncino committed:
PP-1501 Fixup
- Added Cache
- Shortened command params
- Removed comments

matthew-craig-ncino committed:
PP-1501 Removed Deploy-toolkit from font-awesome it wasn't needed.

matthew-craig-ncino committed:
PP-1501 Removed sfdeploy this is NOT a salesforce deployment

matthew-craig-ncino committed:
PP-1501 Changed order for npm

matthew-craig-ncino committed:
PP-1501 added in npm install

matthew-craig-ncino committed:
PP-1501 Cleanup

matthew-craig-ncino committed:
PP-1501 Added Deploy_toolkit back
- It was needed for notifications so added other things in too

matthew-craig-ncino committed:
PP-1501 Added deploy_toolkit branch var

matthew-craig-ncino committed:
PP-1501 removed temp file

matthew-craig-ncino committed:
PP-1501 Removed stray -

Kevmo92 commented on circle.yml:
```diff
@@ -1,31 +1,30 @@
+      if [ "$DEPLOY_TOOLKIT_BRANCH" == "<DEPLOY_TOOLKIT_BRANCH>" ]; then DEPLOY_TOOLKIT_BRANCH="master"; fi
+      echo "Checking out branch: $DEPLOY_TOOLKIT_BRANCH\n"
+      git clone -b $DEPLOY_TOOLKIT_BRANCH git@github.com:loanlifecycle/deploy_toolkit.git $DEPLOY_TOOLKIT
+    - sudo apt-get install --assume-yes --force-yes ruby-dev npm
```
Do we need to specify npm here with node in the machine section? Could we specify ruby in the machine section as well?

Kevmo92 reviewed (commented):
Machine section questions.

matthew-craig-ncino committed:
PP-1501 Experimental build removed ruby-dev from apt-get line

matthew-craig-ncino committed:
PP-1501 Removed install of npm
- I logged into system and npm in installed by default.
- Don't know why it was not recognized before

matthew-craig-ncino committed:
PP-1501 Fixed npm install
- Removing "sudo" from npm install allows instalation of pkgs.

matthew-craig-ncino commented:
Thought that ruby-dev was required. -> it aparently wasn't.
the install of NPM was only required because it was being called with "sudo" once the sudo was removed the install of npm was no longer required. This shaved 30 second off build.

matthew-craig-ncino committed:
[WIP] PP-1501 Experimental release code

matthew-craig-ncino committed:
[WIP] PP-1501 Added creds temporarily for release push

matthew-craig-ncino committed:
[WIP] PP-1501 installed binary in dependency section

matthew-craig-ncino committed:
[WIP] Removal of token

matthew-craig-ncino committed:
[WIP] PP-1501 Release code down to post process

matthew-craig-ncino committed:
[WIP] PP-1501 Fixed user name

matthew-craig-ncino committed:
[WIP] Fixed user issues for login
- notice GITHUB_USER = circle
- user in command line = loanlifecycle

matthew-craig-ncino committed:
[WIP] PP-1501 Better build

matthew-craig-ncino committed:
[WIP] PP-1501 fixed name tag

matthew-craig-ncino committed:
PP-1501 updated file name

matthew-craig-ncino committed:
PP-1501 moved release code to post section

matthew-craig-ncino committed:
PP-1501 moved release code back AND updated branch to test branch

matthew-craig-ncino committed:
PP-1501 fixed spelling

matthew-craig-ncino committed:
PP-1501 Getting branch ready for merge with master

matthew-craig-ncino committed:
PP-1501 Removed env vars from script

Kevmo92 reviewed (approved):
Changes look awesome! The binary is being tagged properly in this repo.
